### ISSUE TITLE
`cached-instance-method (B019)` gives false positive when decorated method belongs to `enum.Enum`

### ISSUE THREAD

**Issue by jakob-keller** [astral-sh/ruff#9912]
What do you think about the following edge case? I believe this should not be reported, since Enum members are typically low cardinality and are not garbage collected anyway.

```python
# test.py

import enum
import functools


class Foo(enum.Enum):
    ONE = enum.auto()
    TWO = enum.auto()

    @functools.cache
    def bar(self, arg: str) -> str:
        return f"{self} - {arg}"
```

```
$ ruff --version
ruff 0.2.1

$ ruff check --isolated --select B test.py
test.py:9:5: B019 Use of `functools.lru_cache` or `functools.cache` on methods can lead to memory leaks
Found 1 error.
```

**Comment by charliermarsh** [comment]
I think I agree with this, though I'd love other opinions / confirmations that this is safe.

**Comment by charliermarsh** [comment]
Confirmed with Alex and Carl that this should be fine.
